### PR TITLE
Support for sign in with local credentials

### PR DESCRIPTION
**Pull request by natalian98**
This PR adds support for signing in with local credentials (kratos `password` method). MFA, recovery and verification will be addressed in future PRs.

## Testing
Set up docker env:
```
docker-compose -f docker-compose.dev.yml up --build --force-recreate --remove-orphans

docker exec <hydra-container> \
  hydra create client \
    --endpoint http://127.0.0.1:4445 \
    --name grafana \
    --grant-type authorization_code,refresh_token \
    --response-type code,id_token \
    --format json \
    --scope openid,offline_access,email \
    --redirect-uri http://localhost:2345/login/generic_oauth

docker run -d --name=grafana -p 2345:2345 --network idp-docker_intranet \
-e "GF_SERVER_HTTP_PORT=2345" \
-e "GF_AUTH_GENERIC_OAUTH_ENABLED=true" \
-e "GF_AUTH_GENERIC_OAUTH_AUTH_ALLOWED_DOMAINS=hydra,localhost" \
-e "GF_AUTH_GENERIC_OAUTH_NAME=Identity Platform" \
-e "GF_AUTH_GENERIC_OAUTH_CLIENT_ID=<client-id>" \
-e "GF_AUTH_GENERIC_OAUTH_CLIENT_SECRET=<client-secret>" \
-e "GF_AUTH_GENERIC_OAUTH_SCOPES=openid offline_access email" \
-e "GF_AUTH_GENERIC_OAUTH_AUTH_URL=http://localhost:4444/oauth2/auth" \
-e "GF_AUTH_GENERIC_OAUTH_TOKEN_URL=http://hydra:4444/oauth2/token" \
-e "GF_AUTH_GENERIC_OAUTH_API_URL=http://hydra:4444/userinfo" \
grafana/grafana

docker exec <kratos-container> kratos import identities /etc/config/kratos/identity.json --endpoint http://127.0.0.1:4434
```

Run login ui:
```
make npm-build build

export KRATOS_PUBLIC_URL="http://localhost:4433"
export HYDRA_ADMIN_URL="http://localhost:4445"
export BASE_URL="http://localhost:4455"
export PORT="4455"
export TRACING_ENABLED="false"
export LOG_LEVEL="debug"
export AUTHORIZATION_ENABLED="false"

go run . serve
```

Go to http://localhost:2345 and sign in with identity platform. It will redirect to login ui and display the option to sign in with local credentials:
image

If incorrect credentials are typed, error will be shown:
image

Logging in with the imported test user (test@example.com / mQ#v9fwHc0@8) logs the user to grafana:
image

I also tested logging with external idp to make sure the changes don't break it.

### CODE REVIEW

**Review by nsklikas — COMMENTED** [review]
Good job overall. I left some comments, but most of them don't require changes in the current PR. To sum up the changes that I think are needed in this PR:
- I think we can get rid of the `UiErrorMessages` struct, we can use `kCLient.LoginFlow` instead
- Would it make sense to just propagate the error description that kratos sends us in `getUiError` instead of creating our own messages? This is part of a larger error handling refactoring we need to do in the login UI, so feel free to disregard it.
- Could we automatically import the Kratos users when running the docker compose? (perhaps we could add an entrypoint script or run "job" container like we do for the migration?)
- Add a kratos/service test that tests the service behavior when we get a kratos error response when using the password method, e.g. the user has entered a wrong password

File: pkg/kratos/service.go [hunk 1]
```diff
@@ -152,6 +175,32 @@ func (s *Service) UpdateOIDCLoginFlow(
 	return &returnToResp, resp.Cookies(), nil
 }
 
+func (s *Service) getUiError(responseBody io.ReadCloser) (err error) {
```
[the remark this answers is not in the document]
Comment (natalian98) [reply pkg/kratos/service.go:178]: We could use it, but the messages are much more detailed and I'm not sure we want to display them to users. For example error code 4000006 stands for `The provided credentials are invalid, check for spelling mistakes in your password or username, email address, or phone number.` We are not going to use phone numbers or usernames as user identifier, so I think it makes sense to shorten the message.

File: docker/kratos/identity.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "traits": {
+      "email": "test@example.com",
+      "name": "Test",
+      "surname": "Example"
+    },
+    "schema_id": "default",
+    "credentials": {
+      "password": {
+        "config": {
+          "password": "mQ#v9fwHc0@8"
```
[the remark this answers is not in the document]
Comment (natalian98) [reply docker/kratos/identity.json]: Done - the reason I used a "difficult" password was to comply with the default password policy, but apparently it is not enforced when importing identities and only during self-service registration.

File: pkg/kratos/service.go [hunk 3]
```diff
@@ -183,6 +232,13 @@ func (s *Service) CheckAllowedProvider(ctx context.Context, loginFlow *kClient.L
 	return s.contains(allowedProviders, fmt.Sprintf("%v", provider)), nil
 }
 
+func (s *Service) getProviderName(updateFlowBody *kClient.UpdateLoginFlowBody) string {
+	if updateFlowBody.GetActualInstance() == updateFlowBody.UpdateLoginFlowWithOidcMethod {
+		return updateFlowBody.UpdateLoginFlowWithOidcMethod.Provider
+	}
+	return ""
```
[the remark this answers is not in the document]
Comment (natalian98) [reply pkg/kratos/service.go:221]: https://github.com/canonical/identity-platform-login-ui/issues/234

File: pkg/kratos/service.go [hunk 4]
```diff
@@ -152,6 +156,33 @@ func (s *Service) UpdateOIDCLoginFlow(
 	return &returnToResp, resp.Cookies(), nil
 }
 
+func (s *Service) getUiError(responseBody io.ReadCloser) (err error) {
+	const IncorrectCredentials = 4000006
+	const InactiveAccount = 4000010
```
Comment (shipperizer) [inline pkg/kratos/service.go:161]: const go at the top of the file, just below imports

**Review by shipperizer — APPROVED** [review]

**Pull request merged**